src/interface/types.rs

```diff
@@ -571,3 +611,43 @@ pub fn get_duration_from_millis(union_argument: Arg) ->Result<Option<interface::
 pub fn arg_nullity(union_argument: &Arg) -> bool {
     unsafe{union_argument.dispatch_cbuf}.is_null()
 }
+
+pub fn get_sigactionstruct<'a>(union_argument: Arg) -> Result<Option<&'a mut SigactionStruct>, i32> {
+    let pointer = unsafe{union_argument.dispatch_sigactionstruct};
+
+    if !pointer.is_null() {
+        Ok(Some(unsafe{&mut *pointer}))
+    } else {
+        Ok(None)
+    }
+}
+
+pub fn get_constsigactionstruct<'a>(union_argument: Arg) -> Result<Option<&'a SigactionStruct>, i32> {
+    let pointer = unsafe{union_argument.dispatch_constsigactionstruct};
+
+    if !pointer.is_null() {
+        Ok(Some(unsafe{& *pointer}))
+    } else {
+        Ok(None)
+    }
+}
+
+pub fn get_sigsett<'a>(union_argument: Arg) -> Result<Option<&'a mut SigsetType>, i32> {
+    let pointer = unsafe{union_argument.dispatch_sigsett};
+
+    if !pointer.is_null() {
+        Ok(Some(unsafe{&mut *pointer}))
+    } else {
+        Ok(None)
+    }
+}
+
+pub fn get_constsigsett<'a>(union_argument: Arg) -> Result<Option<&'a SigsetType>, i32> {
+    let pointer = unsafe{union_argument.dispatch_constsigsett};
+
+    if !pointer.is_null() {
+        Ok(Some(unsafe{& *pointer}))
+    } else {
+        Ok(None)
+    }
+}
```

Comment:
Same as previous cases, the else block after the if isn't necessary. Since you're returning from inside the if block, you can just have the Ok(None) as the next statement.